examples/SuperResolution/SuperResolution.cpp

```diff
@@ -0,0 +1,122 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "examples/SuperResolution/SuperResolution.h"
+
+#include "common/Log.h"
+
+SuperResolution::SuperResolution() : ExampleBase() {
+}
+
+const wnn::Operand SuperResolution::BuildConstantFromNpy(const wnn::GraphBuilder& builder,
+                                                         const std::string& path) {
+    const cnpy::NpyArray data = cnpy::npy_load(path);
+    mConstants.push_back(data.data_holder);
+    return utils::BuildConstant(builder, data.shape, data.data<float>(), data.num_bytes());
+}
+
+const wnn::Operand SuperResolution::BuildConv(const wnn::GraphBuilder& builder,
+                                              const wnn::Operand& input,
+                                              int32_t convIndex,
+                                              bool relu,
+                                              utils::Conv2dOptions* options,
+                                              const std::string& biasName) {
+    std::string prefix = mLayout == "nchw" ? mWeightsPath + "conv" : mWeightsPath + "Const_";
+    std::string suffix = mLayout == "nchw" ? "_weight.npy" : ".npy";
+    const std::string weightsPath = prefix + std::to_string(convIndex) + suffix;
+    const wnn::Operand convWeights = BuildConstantFromNpy(builder, weightsPath);
+
+    // TODO: Figure out correct "channels last" path suffix.
+    prefix = mLayout == "nchw" ? mWeightsPath + "conv" : mWeightsPath + "super_resolution_";
+    if (mLayout == "nchw") {
+        prefix.append(std::to_string(convIndex));
+    }
+
+    const std::string biasPath = prefix + biasName + "_bias.npy";
+    const wnn::Operand convBias = BuildConstantFromNpy(builder, biasPath);
+
+    const wnn::Conv2dOptions* conv2dOptions = options != nullptr ? options->AsPtr() : nullptr;
+    const wnn::Operand conv2d = builder.Conv2d(input, convWeights, conv2dOptions);
+
+    if (!mFused) {
+        if (relu) {
+            return builder.Relu(conv2d);
+        }
+        return conv2d;
+    }
+
+    // Fused
+    utils::Conv2dOptions fusedOptions;
+    if (options != nullptr) {
+        fusedOptions = *options;
+    }
+    fusedOptions.bias = convBias;
+
+    if (relu) {
+        fusedOptions.activation = builder.ReluOperator();
+    }
+
+    return builder.Conv2d(input, convWeights, fusedOptions.AsPtr());
+}
+
+const wnn::Operand SuperResolution::LoadNchw(const wnn::GraphBuilder& builder, bool softmax) {
+    const wnn::Operand input = utils::BuildInput(builder, "input", {1, 1, 224, 224});
+
+    // TODO: Does kernel_shape (5,5) need to be specified?
```

Comment:
The shape is got from npy files seeing L.26.

Reply:
Ack